internal/component/pyroscope/relabel/relabel.go

```diff
@@ -0,0 +1,337 @@
+}
+
+// DefaultArguments provides the default arguments for the pyroscope.relabel
+// component.
+var DefaultArguments = Arguments{
+	MaxCacheSize: 10_000,
+}
+
+// SetToDefault implements syntax.Defaulter.
+func (a *Arguments) SetToDefault() {
+	*a = DefaultArguments
+}
+
+// Exports holds values which are exported by the pyroscope.relabel component.
+type Exports struct {
+	Receiver pyroscope.Appendable `alloy:"receiver,attr"`
+	Rules    alloy_relabel.Rules  `alloy:"rules,attr"`
+}
+
+// Component implements the pyroscope.relabel component.
+type Component struct {
+	opts         component.Options
+	metrics      *metrics
+	mut          sync.RWMutex
+	rcs          []*relabel.Config
+	fanout       *pyroscope.Fanout
+	cache        *lru.Cache
+	maxCacheSize int
+	exited       atomic.Bool
+}
+
+var (
+	_ component.Component = (*Component)(nil)
+)
+
+// New creates a new pyroscope.relabel component.
+func New(o component.Options, args Arguments) (*Component, error) {
+	cache, err := lru.New(args.MaxCacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Component{
+		opts:         o,
+		metrics:      newMetrics(o.Registerer),
+		cache:        cache,
+		maxCacheSize: args.MaxCacheSize,
+	}
+
+	c.fanout = pyroscope.NewFanout(args.ForwardTo, o.ID, o.Registerer)
+
+	o.OnStateChange(Exports{
+		Receiver: c,
+		Rules:    args.RelabelConfigs,
+	})
+
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+func (c *Component) Run(ctx context.Context) error {
+	defer c.exited.Store(true)
+	<-ctx.Done()
+	return nil
+}
+
+func (c *Component) Update(args component.Arguments) error {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+
+	newArgs := args.(Arguments)
+	newRCS := alloy_relabel.ComponentToPromRelabelConfigs(newArgs.RelabelConfigs)
+
+	// If relabeling rules changed, purge the cache
+	if relabelingChanged(c.rcs, newRCS) {
+		level.Debug(c.opts.Logger).Log("msg", "received new relabel configs, purging cache")
+		c.cache.Purge()
+		c.metrics.cacheSize.Set(0)
+	}
+
+	if newArgs.MaxCacheSize != c.maxCacheSize {
+		evicted := c.cache.Resize(newArgs.MaxCacheSize)
+		if evicted > 0 {
+			level.Debug(c.opts.Logger).Log("msg", "resizing cache led to evicting items", "evicted_count", evicted)
+		}
+		c.maxCacheSize = newArgs.MaxCacheSize
+	}
+
+	c.rcs = newRCS
+	c.fanout.UpdateChildren(newArgs.ForwardTo)
+
+	c.opts.OnStateChange(Exports{
+		Receiver: c,
+		Rules:    newArgs.RelabelConfigs,
+	})
+
+	return nil
+}
+
+type ProfileDroppedError struct{}
+
+func (e *ProfileDroppedError) Error() string {
+	return "profile dropped by relabel rules"
+}
+
+func (c *Component) Append(ctx context.Context, lbls labels.Labels, samples []*pyroscope.RawSample) error {
+	if c.exited.Load() {
+		return fmt.Errorf("%s has exited", c.opts.ID)
+	}
+
+	c.mut.RLock()
+	defer c.mut.RUnlock()
+
+	newLabels, keep, err := c.relabel(lbls)
+	if err != nil {
+		return err
+	}
+	if !keep {
+		return &ProfileDroppedError{}
+	}
```

Comment:
Returning an error here feels wrong to me. We also would have to handle this case in all other components that "Append" profiles. And the handling would be at most log, what happend and continue. 

I think we could consider a debug log here, but should not return an error.